### PR TITLE
Refactor Python profiling installation and configuration process

### PR DESCRIPTION
**Pull request by jd**
### What does this PR do?

This refactor the Python installation guide to be clearer and more concise.

### Motivation

It's hard for come users to install the Python profiler with current documentation.

### Preview
https://docs-staging.datadoghq.com/jd/jd/python-profiling-improvement/tracing/profiler/getting_started/?code-lang=python
### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by jd** [comment]
It seems the language tabs are not working in the preview. Is this a bug in the preview or the PR breaking them in a way I don't understand?

**Comment by apigirl** [comment]
> It seems the language tabs are not working in the preview. Is this a bug in the preview or the PR breaking them in a way I don't understand?

It looks like you're missing the closing tags. I'll add a suggestion to correct

**Comment by apigirl** [comment]
> It seems the language tabs are not working in the preview. Is this a bug in the preview or the PR breaking them in a way I don't understand?

So we have a known issue that the code tabs don't support headers yet (so ## and ###). We're working on a fix. For now the path of least resistance would be to update the headers to be bold instead.

**Comment by jd** [comment]
@davidejones why/what did you force-push this branch?

File: content/en/tracing/profiler/getting_started.md [hunk 1]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
 {{< /programming-lang >}}
 {{< programming-lang lang="python" >}}
 
-The Datadog Profiler requires Python 2.7+. Memory profiling is available on Python 3.5+. To begin profiling applications:
+### Requirements
 
-1. If you are already using Datadog, upgrade your agent to version [7.20.2][1]+ or [6.20.2][1]+.
+The Datadog Profiler requires Python 2.7+ and agent to version [7.20.2][1]+ or
```
Comment (cswatt) [inline content/en/tracing/profiler/getting_started.md:81]:
```suggestion
The Datadog Profiler requires Python 2.7+ and Agent version [7.20.2][1]+ or
```

File: content/en/tracing/profiler/getting_started.md [hunk 2]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
+[6.20.2][1]+.
 
-2. Install `ddtrace` which contains both tracing and profiler:
+Those profiling features are available depending on your Python version:
```
Comment (cswatt) [inline content/en/tracing/profiler/getting_started.md:84]:
```suggestion
The following profiling features are available depending on your Python version:
```

File: content/en/tracing/profiler/getting_started.md [hunk 3]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
 
-    ```shell
-    pip install ddtrace
-    ```
+|      Feature         | Supported Python Versions          |
```
Comment (cswatt) [inline content/en/tracing/profiler/getting_started.md:86]:
```suggestion
|      Feature         | Supported Python versions          |
```

File: content/en/tracing/profiler/getting_started.md [hunk 4]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
+| CPU time profiling   | Python >= 2.7 on POSIX platforms   |
+| Exception profiling  | Python >= 3.7 on POSIX platforms   |
+| Lock profiling       | Python >= 2.7                      |
+| Memory profiling     | Python >= 3.5                      |
 
-     **Note**: Profiler is available in the `ddtrace` library for versions 0.36+.
+### Installation
 
-3. To automatically profile your code, set `DD_PROFILING_ENABLED` environment variable to `true` when you use `ddtrace-run`:
+Install `ddtrace` which provides both tracing and profiling functionalities:
```
Comment (cswatt) [inline content/en/tracing/profiler/getting_started.md:96]:
```suggestion
Install `ddtrace`, which provides both tracing and profiling functionalities:
```

File: content/en/tracing/profiler/getting_started.md [hunk 5]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
+
+```shell
+pip install ddtrace
+```
+
+**Note**: Profiling requires the `ddtrace` library versions 0.40+.
```
Comment (cswatt) [inline content/en/tracing/profiler/getting_started.md:102]:
```suggestion
**Note**: Profiling requires the `ddtrace` library version 0.40+.
```

File: content/en/tracing/profiler/getting_started.md [hunk 6]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
+available, you will need to install a development environment.
+
+For example, on Alpine Linux, this can be done with:
+```shell
+apk install gcc musl-dev linux-headers
+```
+
+### Usage
+
+To automatically profile your code, set `DD_PROFILING_ENABLED` environment
```
Comment (cswatt) [inline content/en/tracing/profiler/getting_started.md:114]:
```suggestion
To automatically profile your code, set the `DD_PROFILING_ENABLED` environment
```

File: content/en/tracing/profiler/getting_started.md [hunk 7]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
+variable to `true` when you use `ddtrace-run`:
 
-    ```
     DD_PROFILING_ENABLED=true ddtrace-run python app.py
-    ```
-    **Note:** `DD_PROFILING_ENABLED` is only supported in `dd-trace` version 0.40+. Use the alternate method if you are running an older version of `dd-trace`.
 
-    **Alternate method**
+It is strongly recommended to add tags like `service` or `version` as it
```
Comment (cswatt) [inline content/en/tracing/profiler/getting_started.md:119]:
```suggestion
It is strongly recommended that you add tags like `service` or `version`, as they
```

**Review by cswatt — CHANGES_REQUESTED** [review]
grammar changes

File: content/en/tracing/profiler/getting_started.md [hunk 8]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
-    ```
+If you want to manually control the lifecycle of the profiler, use the
+`ddtrace.profiling.profiler.Profiler` object:
 
-4. After a minute or two, visualize your profiles on the [Datadog APM > Profiler page][2].
+```python
+from ddtrace.profiling import Profiler
 
-- It is strongly recommended to add tags like `service` or `version` as it provides the ability to slice and dice your profiles across these dimensions, enhancing your overall product experience. Use environment variables to set the parameters:
+prof = Profiler()
```
Comment (dbenamydd) [inline content/en/tracing/profiler/getting_started.md:131]: And we could do the equivalent here, with comments showing what env var is used if not specified, like
```
prof = Profiler(
    env="prod",  # if not specified, uses env var DD_ENV
...
```

File: content/en/tracing/profiler/getting_started.md [hunk 9]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
+# At shutdown
+prof.stop()
+```
 
-<div class="alert alert-info">
-Recommended for advanced usage only.
+<div class="alert alert-warning">
+The profiler has been designed to be turned on continuously. Only use the
+`start` and `stop` methods to start and stop the profiler at the beginning and
+end of your program, not to selectively profile parts of your program.
```
Comment (dbenamydd) [inline content/en/tracing/profiler/getting_started.md:141]: If we do the above, we can probably ditch this. The less stuff people have to think about when onboarding, the better!

File: content/en/tracing/profiler/getting_started.md [hunk 10]
```diff
@@ -76,70 +76,103 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
-    ```
+You can configure the profiler using the following environment variables:
+
+| Environment variable                             | Type          | Description                                                                                      |
+| ------------------------------------------------ | ------------- | ------------------------------------------------------------------------------------------------ |
+| `DD_PROFILING_ENABLED`                           | Boolean       | Set to `true` to enable profiler. |
+| `DD_SERVICE`                                     | String        | The Datadog [service][3] name.     |
+| `DD_ENV`                                         | String        | The Datadog [environment][4] name, for example, `production`. |
+| `DD_VERSION`                                     | String        | The version of your application.                             |
+| `DD_TAGS`                                        | String        | Tags to apply to an uploaded profile. Must be a list of `<key>:<value>` separated by commas such as: `layer:api,team:intake`.   |
```
Comment (dbenamydd) [inline content/en/tracing/profiler/getting_started.md:175]: Work another col for the programmatic equivalent of each?

**Review by dbenamydd — COMMENTED** [review]
This is much cleaner structure and generally better!

File: content/en/tracing/profiler/getting_started.md [hunk 11]
```diff
@@ -76,70 +76,102 @@ The Datadog Profiler requires [JDK Flight Recorder][1]. The Datadog Profiler lib
...
+    DD_PROFILING_ENABLED=true \
+    DD_ENV=prod \
+    DD_SERVICE=my-web-app \
+    DD_VERSION=1.0.3 \
+    dd-trace-run python app.py
 
-<div class="alert alert-info">
-Recommended for advanced usage only.
-</div>
+It is strongly recommended that you add tags like `service` or `version`, as they
```
Comment (dbenamydd) [inline content/en/tracing/profiler/getting_started.md:123]: I wonder if folks will think they need to do something in addition to the env vars above. Maybe something like

> In addition to the env, service, and version shown above, you can add other tags like `DD_TAGS=canary:true,shard:3` which will provide the ability to slice and dice...

Or maybe we just mention DD_TAGS and leave details for below. I'm not sure, just throwing something out to make it concrete. Non-blocking.

**Review by dbenamydd — APPROVED** [review]
This is awesome!

**Comment by jd** [comment]
@cswatt @DataDog/corpweb what's blocking this from being merged?

**Review by GaurabAryal — APPROVED** [review]

**Review by cswatt — APPROVED** [review]

**Pull request merged**